lib/smart_proxy_remote_execution_ssh.rb

```diff
@@ -83,6 +84,11 @@ def validate_ssh_log_level!
         Plugin.settings.ssh_log_level = Plugin.settings.ssh_log_level.to_sym
       end
 
+      def validate_socket_path!
+        socket_path = Plugin.settings.socket_working_dir
+        raise "Socket path #{socket_path} too long" if socket_path.length > Plugin::SOCKET_PATH_MAX_LENGTH
```

Comment:
It should expand the path before checking the length. On my machine `~/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa` resolves to `/home/aruzicka/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`. One passes this check, the other does not.